standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -725,7 +725,7 @@ struct SetPartitionsStatsRequest {
 2: optional bool needMerge, //stats need to be merged with the existing stats
 3: optional i64 writeId=-1,         // writeId for the current query that updates the stats
 4: optional string validWriteIdList, // valid write id list for the table for which this struct is being sent
-5: required string engine //engine creating the current request
+5: optional string engine = "hive" //engine creating the current request
```

Comment:
Make the `engine` field optional, and give it a default value `hive` which is consistent with the current existing code logic.

Reply:
could we set the `engine` to default value `hive` without changing it to `optional`?

Reply:
That doesn't work. `required `will enforce check `engine `field in construction method though it has a default value. For some others framework(Trino) will use this class `SetPartitionsStatsRequest` but don't want to care the `engine `field, and they just reuse hive stats.

`optional `has no any problem in current hive code. only need partial trival changes in `HiveMetaStoreClient.java`. I think it is safe. :)

Reply:
Though I am not familiar with Thrift, I expect the following behaviors.

| Client protocol | HMS version | Behavior |
|-|-|-|
| Hive 3 (no engine is set) | Hive 4 w/o HIVE-27984 | Error |
| Hive 4 w/o HIVE-27984 | Hive 4 w/o HIVE-27984 | The engine type specified by the client is used |
| Hive 4 w/ HIVE-27984 | Hive 4 w/o HIVE-27984 | The engine type specified by the client is used |
| Hive 3 (no engine is set) | Hive 4 w/ HIVE-27984 | engine=hive |
| Hive 4 w/o HIVE-27984 | Hive 4 w/ HIVE-27984 | The engine type specified by the client is used |
| Hive 4 w/ HIVE-27984 | Hive 4 w/ HIVE-27984 | The engine type specified by the client is used |

Reply:
Yes, you are almost right. To be more specific, for example, Trino will get hms table column statistics by its thrift hms client which is compatible with hive3.
https://github.com/trinodb/trino/blob/057f2d435d6461855ff0f904dbd675b39db97d1e/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java#L331-L336

But `TableStatsRequest` in Hive4 need pass `engine` through constructor function, and the `engine` field is required, therefore:

1. for hive3 thrift client, this api call will fail due to the missed `engine `field if HMS server doesn't have this PR, As there is no place for hive3 thrift client to pass `engine `field. But this api call will sucessfully if HMS has this PR(`engine` field is optional and has the defalut value).
2. for hive4 thrift client without this PR, you also must pass `engine `field in `TableStatsRequest` to call this api correctly no matter if HMS server has this PR. Because `engine` field is needed in `TableStatsRequest` constructor function.
3. for hive4 thrift client with this PR. Whether HMS server have this PR or not
, you can call the api sucessfully without pass `engine` as the `engine` field is not mandatory in `TableStatsRequest` constructor function, and you can explicitly reset the `engine` fileld by `TableStatsRequest::setEngine` if you don't want the default `hive engine` column statistics. But i think most  framework(like trino) will want reuse hive engine column statistics.

Sounds a little verbose.. :) In one word, this PR can make hive3 thrift hms client worked well with hive4 hms when calling statistics related api. I have tested this PR with Trino, and i think it is good.

Reply:
Thanks for elaborating the behaviors. I'd say it is flexible and less surprising for people.